tools/src/launcher/Main.hs

```diff
@@ -515,11 +523,73 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
-    logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
+    customLogger LInfo NWallet "Starting the wallet"
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = createProc PHandle path args (Just logHandle)
+            system' phvar cr mempty NWallet
+        Nothing ->
+           if shouldLog then
+               liftIO $ do
+               let cr = createProc PInherit path args Nothing
+               system' phvar cr mempty NWallet
+           else
+               liftIO $ do
+               let cr = createProc PNoStream path args Nothing
+               system' phvar cr mempty NWallet
+
+createProc :: ProcessType -> FilePath -> [Text] -> (Maybe Handle) -> Process.CreateProcess
```

Comment:
Why do you need `ProcessType` type at all? Why not just pass `StdStream` from `System.Process`? The only different between these types is that your `PHandle` doesn't have `Handle` inside, but `UseHandle` from `StdStream` does. However, whenever you pass `PHandle`, you also pass `Just Handle`.
I really don't see any good reason to have `ProcessType` type. Just use `StdStream`.

Reply:
If you pass `StdStream`, this function can be written much-much shorter.
```
createProc stdStream path args = (Process.proc path (map toString args))
  { Process.std_in = Process.CreatePipe
  , Process.std_out = stdStream
  , Process.std_err = stdStream
  }
```

Reply:
Done